pinot-filesystem/src/main/java/com/linkedin/pinot/filesystem/PinotFS.java

```diff
@@ -110,6 +110,13 @@
    */
   public abstract void copyFromLocalFile(File srcFile, URI dstUri) throws Exception;
 
+  /**
+   * Allows us the ability to determine whether the uri is a directory.
+   * @param uri location of file or directory
+   * @return true if uri is a directory, false otherwise.
+   */
+  public abstract boolean isDirectory(URI uri) throws IOException;
```

Comment:
Why does this need to be a URI?

Reply:
alternative is a string, and URI is better imo. Also consistent with the other apis